### PR TITLE
feat(flutter): add actions to replace flutter scripts

### PR DESCRIPTION
**Pull request by shkcodes**
This PR adds a few Flutter actions which were earlier being accomplished using scripts in the Flutter template. It's companion PR can be found [here](https://github.com/Fueled/project-template-flutter/pull/32) which removes the scripts from the Flutter template.

### CODE REVIEW

File: lib/fastlane/plugin/fueled/actions/formatting_checks_flutter.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module Fastlane
+  module Actions
+    module SharedValues
+    end
+
+    class FormattingChecksFlutterAction < Action
+      def self.run(params)
+        puts __dir__
```
Comment (notbenoit) [inline lib/fastlane/plugin/fueled/actions/formatting_checks_flutter.rb:8]: Looks like you left a `puts` here. Maybe use `UI.message` or remove it if it was for debug purposes ?

**Review by notbenoit — APPROVED** [review]
Looks good 👍  Pre approving !

File: lib/fastlane/plugin/fueled/actions/formatting_checks_flutter.rb (hunk 1 above)
Comment (shkcodes) [reply lib/fastlane/plugin/fueled/actions/formatting_checks_flutter.rb:8]: oops, good catch! 🕵️

File: lib/fastlane/plugin/fueled/actions/formatting_checks_flutter.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        sh("flutter pub get")
+        sh("flutter format -o none --set-exit-if-changed .")
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Format Flutter project"
```
Comment (humblerookie) [inline lib/fastlane/plugin/fueled/actions/formatting_checks_flutter.rb:18]: -> `Check Formatting for a Flutter project`

File: lib/fastlane/plugin/fueled/actions/generate_files_flutter.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      def self.description
+        "Prepare Flutter project"
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :build_variant,
+            env_name: "BUILD_VARIANT",
+            description: "The build variant used for generating files",
```
Comment (humblerookie) [inline lib/fastlane/plugin/fueled/actions/generate_files_flutter.rb:29]: `The build variant used for generating config file`

File: lib/fastlane/plugin/fueled/actions/generate_files_flutter.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        sh("flutter pub run easy_localization:generate -S \"assets/translations\" -O \"lib/gen\"")
+        sh("flutter pub run easy_localization:generate -S \"assets/translations\" -O \"lib/gen\" -o \"locale_keys.g.dart\" -f keys")
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Prepare Flutter project"
```
Comment (humblerookie) [inline lib/fastlane/plugin/fueled/actions/generate_files_flutter.rb:21]: -> `Run code generators for a flutter project`

**Review by humblerookie — APPROVED** [review]
Looks good, just a few minor suggestions around descriptions

**Review by husseinala — APPROVED** [review]
Looks good!

**Pull request merged**